test/BinaryParse.cpp

```diff
@@ -249,20 +195,32 @@ ParsedInstruction MakeParsedInt32TypeInstruction(uint32_t result_id) {
   return ParsedInstruction(parsed_i32_inst);
 }
 
-using BinaryParseTest =
-    spvtest::TextToBinaryTestBase<CaptureParseResults<::testing::Test>>;
+class BinaryParseTest : public spvtest::TextToBinaryTestBase<::testing::Test> {
+ protected:
+  void Parse(const SpirvVector& binary, spv_result_t expected_result) {
+    EXPECT_EQ(expected_result,
+              spvBinaryParse(context, &client_, binary.data(), binary.size(),
+                             invoke_header, invoke_instruction, &diagnostic_));
+  }
+
+  spv_diagnostic diagnostic_ = nullptr;
+  MockParseClient client_;
+};
+
+// Adds an EXPECT_CALL to client_->Header() with appropriate parameters,
+// including bound.  Returns the EXPECT_CALL result.
+#define EXPECT_HEADER(bound)                                                 \
+  EXPECT_CALL(client_,                                                       \
+              Header(AnyOf(SPV_ENDIANNESS_LITTLE, SPV_ENDIANNESS_BIG),       \
+                     SpvMagicNumber, SpvVersion,                             \
+                     SPV_GENERATOR_WORD(SPV_GENERATOR_KHRONOS_ASSEMBLER, 0), \
+                     bound, 0 /*reserved*/))
 
 TEST_F(BinaryParseTest, EmptyModuleHasValidHeaderAndNoInstructionCallbacks) {
   const auto binary = CompileSuccessfully("");
-  spv_diagnostic diagnostic = nullptr;
-  EXPECT_EQ(SPV_SUCCESS,
-            spvBinaryParse(context, this, binary.data(), binary.size(), Header,
-                           Instruction, &diagnostic));
-  EXPECT_EQ(nullptr, diagnostic);
-  EXPECT_THAT(endians(), AnyOf(Eq(Endians{SPV_ENDIANNESS_LITTLE}),
-                               Eq(Endians{SPV_ENDIANNESS_BIG})));
-  EXPECT_THAT(headers(), Eq(Sentences{ExpectedHeaderForBound(1)}));
-  EXPECT_THAT(instructions(), Eq(Instructions{}));
+  EXPECT_HEADER(1).WillOnce(Return(SPV_SUCCESS));
+  Parse(binary, SPV_SUCCESS);
+  EXPECT_EQ(nullptr, diagnostic_);
```

Comment:
The old version of the test also ensured that the instruction callback was never called.

So should add 
```
EXPECT_CALL(client_, Instruction(_)).Times(0)
```

I assume we can say Times(0).